### PR TITLE
use property based testing

### LINKED ISSUE #687: Use property based testing
Use proptest in tests that use random values.

https://docs.rs/proptest/latest/proptest/

(Moved from todo comment to issue in https://github.com/scroll-tech/ceno/pull/659/commits/705a7a33fdb1bbdb725074280cf0854141bd16fc)

### PR DESCRIPTION
**Pull request by zemse**
Works towards #687

### CODE REVIEW

File: ceno_zkvm/src/instructions/riscv/slt.rs [hunk 1]
```diff
@@ -244,10 +246,9 @@ mod test {
 
     #[test]
     fn test_sltu_random() {
-        let mut rng = rand::thread_rng();
-        let a: u32 = rng.gen();
-        let b: u32 = rng.gen();
-        verify::<SltuOp>("random 1", a, b, (a < b) as u32);
-        verify::<SltuOp>("random 2", b, a, (a >= b) as u32);
+        proptest::proptest!(|(a in any::<u32>(), b in any::<u32>())| {
```
Comment (matthiasgoergens) [inline ceno_zkvm/src/instructions/riscv/slt.rs:249]: Thanks for taking this on!

Please see https://github.com/scroll-tech/ceno/pull/667 for something similar.

The main difference is `any::<u32>` picks from all `u32` uniformly at random, but that PR is slightly more clever about the sampling.

**Comment by matthiasgoergens** [comment]
> Closes #687

I'd like that issue to cover all uses of random generation in our tests, not just this particular example, please.

File: ceno_zkvm/src/instructions/riscv/slt.rs (hunk 1 above)
Comment (zemse) [reply ceno_zkvm/src/instructions/riscv/slt.rs:249]: Oh I missed looking at that, sorry

Comment (matthiasgoergens) [reply ceno_zkvm/src/instructions/riscv/slt.rs:249]: No worries. I didn't advertise enough, I guess.

**Pull request closed without merging**